bitwarden_license/src/Commercial.Core/SecretsManager/Commands/Secrets/DeleteSecretCommand.cs

```diff
@@ -30,9 +38,29 @@ public DeleteSecretCommand(ISecretRepository secretRepository)
             {
                 throw new NotFoundException();
             }
-            // TODO Once permissions are implemented add check for each secret here.
             else
             {
+                var hasAccess = false;
+
+                if(secret.Projects == null || secret.Projects?.Count == 0){
+                    hasAccess = orgAdmin;
+                } else {
+
+                    var projectId = secret.Projects.FirstOrDefault().Id;
+
+                    hasAccess = accessClient switch
+                    {
+                        AccessClientType.NoAccessCheck => true,
+                        AccessClientType.User => false,
+                        _ => false,
+                    };
+                }
+
+                if (!hasAccess)
+                {
+                    throw new UnauthorizedAccessException();
```

Comment:
Not sure about the type of exception we want to throw in these cases, and if we want it to be an exception at all..?